include/libtorrent/file.hpp

```diff
@@ -187,6 +187,14 @@ namespace libtorrent
 // internal export should be used at unit tests only
 	TORRENT_EXTRA_EXPORT native_path_string convert_to_native_path_string(std::string const& path);
 
+// internal export should be used at unit tests only
+	TORRENT_EXTRA_EXPORT std::string convert_from_native_path(char const* s);
```

Comment:
checked vanilla MinGW (https://sourceforge.net/projects/mingw/) and new MinGW (https://sourceforge.net/projects/mingw-w64/).
old MinGW just could not compile libTorrent due lack of c++11 support (std::thread not found, std::snprintf not found and so on).
MinGW-w64 works well out of box (with little tweak for sha1,sha512 source support). 
Removing workaround for CWD gives no error and test_file passed OK.